weather_dl/download_pipeline/parsers.py

```diff
@@ -303,12 +381,16 @@ def require(condition: bool, message: str) -> None:
 
             Supported clients are {}
             """.format(str(list(CLIENTS.keys()))))
-    if params.get('append_date_dirs', 'false') == 'true':
-        require(use_date_as_directory(config),
-                """
-                'append_date_dirs' set to true, but creating the date directory hierarchy also
-                requires that 'target_filename' is given and that 'date' is a partition_key.
-                """)
+    if 'append_date_dirs' in params:
+        raise NotImplementedError("Current version of weather-tool is no longer supporting 'append_date_dirs'! "
+                                  "Please refer to documentation for creating date-based directory hierarchy "
+                                  "(https://github.com/google/weather-tools/blob/main/Configuration.md"
+                                  "#creating-a-date-based-directory-hierarchy).")
+    if 'target_filename' in params:
+        raise NotImplementedError("Current version of weather-tool is no longer supporting 'target_filename'! "
+                                  "Please refer to documentation "
+                                  "(https://github.com/google/weather-tools/blob/main/Configuration.md"
+                                  "#parameters-section).")
```

Comment:
How about we make use of the `require` method so that we have consistent string formatting? I do think it's a good idea to use `NotImplementedError`s. Let's update `require` to take an optional argument that is the type of error (defaulting to ValueError).